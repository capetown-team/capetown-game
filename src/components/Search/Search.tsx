import React, { memo, FC } from 'react';
import block from 'bem-cn-lite';
import { InputType } from '@/types.d';

import './Search.scss';

const b = block('search');

export type Props = {
  searchHandler: InputType;
};

const Search: FC<Props> = ({ searchHandler }) => (
  <div className={b()}>
    <input
      className={b('input')}
      onChange={searchHandler}
      type="text"
      placeholder="Поиск"
    />
  </div>
);

const WrappedSearch = memo(Search);

export { WrappedSearch as Search };
