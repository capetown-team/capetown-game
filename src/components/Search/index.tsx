export { Search } from '@/components/Search/Search';
