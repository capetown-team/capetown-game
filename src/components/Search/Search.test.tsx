import React, { ChangeEvent } from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import { Search } from './Search';

describe('<Search />', () => {
  it('should renders correct <Button />', () => {
    const value = { text: '' };
    const wrapper = shallow(
      <Search
        searchHandler={(event: ChangeEvent<HTMLInputElement>) => {
          value.text = event.target.value;
        }}
      />
    );

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
