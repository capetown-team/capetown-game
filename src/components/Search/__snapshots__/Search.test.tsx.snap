// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> should renders correct <Button /> 1`] = `
<div
  className="search"
>
  <input
    className="search__input"
    onChange={[Function]}
    placeholder="Поиск"
    type="text"
  />
</div>
`;
